src/serializer/Emitter.js

```diff
@@ -129,25 +147,30 @@ export class Emitter {
     while (a.length > 0) {
       let { body, dependencies, func } = a.shift();
       if (body !== oldBody) {
-        invariant(this._activeBodies.has(body));
-        let b = this._waitingForBodies.get(body);
-        if (b === undefined) this._waitingForBodies.set(body, (b = []));
-        b.push({ dependencies, func });
+        this._emitAfterWaitingGeneratorBody(body, dependencies, func);
       } else {
         this.emitNowOrAfterWaitingForDependencies(dependencies, func);
       }
     }
     this._waitingForValues.delete(value);
   }
 
+  // Find the first ancestor in input stack that is in current active stack.
+  // It can always find one because the bottom one in the stack is the main generator.
+  _getFirstAncestorGeneratorWithActiveBody(bodyStack: Array<GeneratorBody>): GeneratorBody {
+    const activeBody = bodyStack.slice().reverse().find(body => this._activeBodies.includes(body));
+    invariant(activeBody);
+    return activeBody;
+  }
+
   // Serialization of a statement related to a value MUST be delayed if
   // the creation of the value's identity requires the availability of either:
   // 1. a time-dependent value that is declared by some generator entry
   //    that has not yet been processed
   //    (tracked by the `_declaredAbstractValues` set), or
   // 2. a value that is also currently being serialized
   //    (tracked by the `_activeStack`).
```

Comment:
Could you update this comment with the case where we have to wait for a generator to become active?